### PR TITLE
autopsy 4.21.0

### PR DESCRIPTION
**Pull request by xakep8**
This commit is part of issue #139929. It upgrades the autopsy formula.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/a/autopsy.rb [hunk 1]
```diff
@@ -1,9 +1,11 @@
 class Autopsy < Formula
   desc "Graphical interface to Sleuth Kit investigation tools"
-  homepage "https://www.sleuthkit.org/autopsy/index.php"
-  url "https://downloads.sourceforge.net/project/autopsy/autopsy/2.24/autopsy-2.24.tar.gz"
-  sha256 "ab787f519942783d43a561d12be0554587f11f22bc55ab79d34d8da703edc09e"
+  homepage "https://www.autopsy.com/"
+  url "https://github.com/sleuthkit/autopsy.git",
+      tag:      "autopsy-4.21.0",
+      revision: "c23b4fd71d8966a80a8d6d2b5d1a979522dfcc49"
```
Comment (carlocab) [inline Formula/a/autopsy.rb:6]: Do we need a git clone? Can we use a checkout instead?

Comment (xakep8) [reply Formula/a/autopsy.rb:6]: @carlocab I could use the zip ig that would be better than a clone, should I do that?

File: Formula/a/autopsy.rb [hunk 2]
```diff
@@ -20,89 +26,48 @@ class Autopsy < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "5e1ce8b5147639d7737a4013030ee2a059d1b8dd4657554e08e9423a9a6b2f66"
   end
 
-  # Installs prebuilt binaries, broken on arm: https://github.com/Homebrew/homebrew-core/issues/175053
-  deprecate! date: "2024-06-22", because: :does_not_build
```
Comment (carlocab) [inline Formula/a/autopsy.rb:24]: Thanks for working on this, but it looks like the reason we deprecated it still applies:
```
  autopsy
    * Binaries built for a non-native architecture were installed into autopsy's prefix.
      The offending files are:
        /opt/homebrew/Cellar/autopsy/4.21.0/libexec/autopsy/markmckinnon/Export_srudb_macos	(x86_64)
        /opt/homebrew/Cellar/autopsy/4.21.0/libexec/autopsy/markmckinnon/parse_prefetch_macos	(x86_64)
        /opt/homebrew/Cellar/autopsy/4.21.0/libexec/platform/modules/lib/x86_64/libjnidispatch-nb.jnilib	(x86_64)
```

Comment (xakep8) [reply Formula/a/autopsy.rb:24]: @carlocab the formula is still being worked on I was just testing what was the issue with the old formula

Comment (carlocab) [reply Formula/a/autopsy.rb:24]: Sure, no problem! Please feel free to continue working on this.

When you feel that this PR is close to ready, please edit your commit message to read
```
autopsy 4.21.0
```

Thanks!

**Pull request closed without merging**